### PR TITLE
[primTorch] refs: masked_fill

### PR DESCRIPTION
**Pull request by kshitij12345**
TODO

* [x] Add error inputs

### CODE REVIEW

**Comment by kshitij12345** [comment]
NOTE: nvFuser failure is real.

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -2417,6 +2418,36 @@ def uniform(
     return prims.uniform(shape, low=low, high=high, dtype=dtype, device=device)
 
 
+def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLikeType):
+    python_type = utils.dtype_to_type(a.dtype)
+    if isinstance(value, Number):
```
Comment (mruberry) [inline torch/_refs/__init__.py:2423]: Let's file an issue for this unconditional cast -- fill_ has similar behavior and I think it's a bug -- then let's comment the issue here

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -2417,6 +2418,36 @@ def uniform(
...
+        value_type = type(value)
+    else:
+        # NOTE: Could not use value = item(value) as it resulted in
+        # RuntimeError: Cannot cast FakeTensor(cpu) to number
+        assert isinstance(value, TensorLike)
+        check(
+            value.ndim == 0,
+            lambda: "only supports a 0-dimensional value tensor, but got tensor with 1 dimension",
```
Comment (mruberry) [inline torch/_refs/__init__.py:2431]: "tensor with X dimensions"

**Review by mruberry — APPROVED** [review]
Approved but see inline suggestions for review

**Pull request closed without merging**